pkg/metrics/metrics.go

```diff
@@ -0,0 +1,231 @@
+package metrics
+
+import (
+	"errors"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+)
+
+type MetricDefinition struct {
+	Name   string
+	Help   string
+	Type   metricType
+	Labels []string
+}
+
+type PromConnectionInfo struct {
+	Address string
+	Port    int
+}
+
+type Settings struct {
+	PromConnectionInfo
+	Prefix string
+}
+
+type metricType string
+
+const (
+	TypeCounter   metricType = "counter"
+	TypeGauge     metricType = "gauge"
+	TypeHistogram metricType = "histogram"
+)
+
+var allMetrics = []MetricDefinition{}
+
+func defineMetric(name, help string, t metricType, labels ...string) MetricDefinition {
+	def := MetricDefinition{
+		Name:   name,
+		Help:   help,
+		Type:   t,
+		Labels: labels,
+	}
+	allMetrics = append(allMetrics, def)
+	return def
+}
+
+var (
+	hmapEvictionsTotal = defineMetric(
+		"hashmap_evictions_total",
+		"Number of hashmap evictions total",
+		TypeCounter,
+		"operational",
+	)
+	userspaceNumberOfEvictionsTotal = defineMetric(
+		"userspace_number_of_evictions_total",
+		"Number of userspace evictions total",
+		TypeCounter,
+		"operational",
+	)
+	numberOfevictedFlowsTotal = defineMetric(
+		"number_of_evicted_flows_total",
+		"Number of evicted flows Total",
+		TypeCounter,
+		"operational",
+	)
+	numberofFlowsreceivedviaRingBufferTotal = defineMetric(
+		"number_of_flows_received_via_ring_buffer_total",
+		"Number of flows received via ring buffer total",
+		TypeCounter,
+		"operational",
+	)
+	timeSpentInLookupandDeleteMapSecondsTotal = defineMetric(
```

Comment:
name suggestion: `lookup_and_delete_map_duration_seconds`
(we don't use the "total" suffix here since we are not adding measurements)